src/status.js

```diff
@@ -1,65 +1,97 @@
-          li.classList.add('list-item');
-          li.innerHTML = text;
-          list.appendChild(li);
-        } else {
-          const li = document.createElement('li');
-          const text = `<div class="list-container"> <input class='check-input' type='checkbox' id='${i}' aria-label='...'>
-            <p class="todo-text">${task.itemText}</p><a class="del-menu" href="#"><i class="fas fa-ellipsis-v"></i></a></div>`;
-          li.classList.add('list-item');
-          li.innerHTML = text;
-          list.appendChild(li);
+
+      const todos = this.data.sort((a, b) => {
+        const indexA = a.todoIndex;
+        const indexB = b.todoIndex;
+
+        if (indexA < indexB) {
+          return -1;
         }
+        if (indexA > indexB) {
+          return 1;
+        }
+        return 0;
+      });
+      const li = document.createElement('li');
+      todos.forEach((task) => {
+        const text = `<div class="list-container main-list"> <input class='check-input' type='checkbox'  aria-label='...'>
+            <p class="todo-text">${task.todoText}</p><a class="del-menu" id="del" href="#"><i class="fas fa-ellipsis-v"></i></a></div>`;
+        li.classList.add('list-item');
+        li.innerHTML = text;
+        this.mainList.appendChild(li);
       });
-    }
+      localStorage.setItem('task', JSON.stringify(todos));
+      this.todoTask.value = '';
+      this.completeTodo();
+      this.removeData();
+    });
   }
 
-  getItems() {
-    return JSON.parse(localStorage.getItem('tasks'));
+  loadData() {
+    window.addEventListener('load', (e) => {
+      e.preventDefault();
+      const tasks = JSON.parse(localStorage.getItem('task'));
+      tasks.forEach((task) => {
+        const li = document.createElement('li');
+        const text = `<div class="list-container main-list"> <input class='check-input' type='checkbox'  aria-label='...'>
+              <p class="todo-text">${task.todoText}</p><a class="del-menu" id="del" href="#"><i class="fas fa-ellipsis-v"></i></a></div>`;
+        li.classList.add('list-item');
+        li.innerHTML = text;
+        this.mainList.appendChild(li);
+      });
+      this.completeTodo();
+      this.removeData();
+    });
   }
 
-  addItems(todo) {
-    localStorage.setItem('tasks', JSON.stringify(todo));
+  removeData() {
+    const del = document.querySelectorAll('#del');
+    del.forEach((v, i) => {
+      v.addEventListener('click', (e) => {
+        e.preventDefault();
+        const div = v.parentElement;
+        const li = div.parentElement;
+        li.remove();
+        const newTask = JSON.parse(localStorage.getItem('task'));
+        newTask.splice(i, 1);
+        localStorage.setItem('task', JSON.stringify(newTask));
+      });
+    });
   }
 
-  updateCheckbox(todoList) {
-    const checkBox = document.querySelectorAll('.check-input');
-    checkBox.forEach((v, i) => {
+  completeTodo() {
```

Comment:
- Add a function for editing a todo item. Make it easier to correct mistakes made in creating a todo item.

Reply:
Can you please check, I already have a function for adding a to-do item? My `completeTodo()` function is only meant for the `<checkbox>.checked` value. Which is returning either true or false.